presto-native-execution/presto_cpp/main/PrestoExchangeSource.cpp

```diff
@@ -319,10 +325,21 @@ PrestoExchangeSource::createExchangeSource(
     int destination,
     std::shared_ptr<exec::ExchangeQueue> queue,
     memory::MemoryPool* pool) {
-  if (strncmp(url.c_str(), "http://", 7) == 0 ||
-      strncmp(url.c_str(), "https://", 8) == 0) {
+  if (strncmp(url.c_str(), "http://", 7) == 0) {
     return std::make_unique<PrestoExchangeSource>(
         folly::Uri(url), destination, queue, pool);
+  } else if (strncmp(url.c_str(), "https://", 8) == 0) {
+    auto systemConfig = SystemConfig::instance();
+    auto clientCertAndKeyPath =
+        systemConfig->httpsClientCertAndKeyPath().value_or("");
+    auto ciphers = systemConfig->httpsClientCertAndKeyPath().value_or("");
+    return std::make_unique<PrestoExchangeSource>(
+        folly::Uri(url),
+        destination,
+        queue,
+        pool,
+        clientCertAndKeyPath,
```

Comment:
Can we cover the https in PrestoExchangeSourceTest.cpp in followup? Thanks!

Reply:
Yeah I will do in the next PR